examples/src/tower/client.rs

```diff
@@ -57,13 +58,9 @@ mod service {
         }
 
         fn call(&mut self, req: Request<BoxBody>) -> Self::Future {
-            let mut channel = self.inner.clone();
```

Comment:
We actually want to keep it this way because it shows how you can embed async work around the future of the channel.

Reply:
There is another solution which we can apply.

```rust
let clone = self.inner.clone();
let mut inner = std::mem::replace(&mut self.inner, clone);
Box::pin(async move {
    // Do extra async work here...
    channel.call(req).await.map_err(Into::into)
})
```

Reply:
@LucioFranco See my comment here https://github.com/tower-rs/tower/issues/547#issuecomment-767629149

Reply:
Oh interesting.....maybe the mem replace works? @olix0r what do you think?

Reply:
Everything is fine?